source/web-contents.lisp

```diff
@@ -207,3 +207,23 @@
                    client.write(`${jsonString}\\n`);})"
              (remote-symbol web-contents) code user-gesture identifier)
      :replace-newlines-p nil)))
+
+(export-always 'on)
+(defmethod on ((web-contents web-contents) event-name code)
+  (send-message-interface
+   (interface web-contents)
+   (format nil "~a.on('~a', () => {~a})" (remote-symbol web-contents) event-name code)
+   :replace-newlines-p nil))
+
+(export-always 'on-event)
+(defmethod on-event ((web-contents web-contents) event-name callback)
+  (let ((identifier (new-integer-id)))
+    (setf (gethash identifier (callbacks electron:*interface*))
+          (lambda (args)
+            (declare (ignore args))
+            (funcall callback web-contents) nil))
```

Comment:
Note that I force the callback to return nil by wrapping it in a lambda. I do this so that `dispatch-callback` doesn't send a message back to the JavaScript side.